examples/BUILD.bazel

```diff
@@ -22,6 +22,11 @@ bazel_integration_test(
     name = "simple_test",
     bazel_version = CURRENT_BAZEL_VERSION,
     test_runner = ":simple_test_runner",
+    env = select({
+    # Linux platforms require that CC be set to clang.
+        "@platforms//os:linux": {"CC": "clang"},
+        "//conditions:default": {},
+    }),
```

Comment:
Curious. Did all of the example integration tests need the `CC=clang` or did you just prophylactically add them to ensure that they would work?

Reply:
I believe they all need it with current CI setup because gnu toolchain is missing there. Maybe it would be more clean to just add proper toolchain in linux image on CI

Reply:
Perhaps, we should consider a different tact. What if we add a parameter that controls whether we add the `CC=clang`? For cases when you don't want to do so, you can toggle the parameter.

The following is some pseudo-code that tries to demonstrate what I am thinking.

```python
def bazel_integration_test(
    name,
    test_runner,
    bazel_version = None,
    bazel_binary = None,
    workspace_path = None,
    workspace_files = None,
    tags = integration_test_utils.DEFAULT_INTEGRATION_TEST_TAGS,
    timeout = "long",
    env = {},
    env_inherit = _DEFAULT_ENV_INHERIT,
    additional_env_inherit = [],
    add_cc_env_for_linux = True,
    **kwargs):

    # Merge env, if necessary
    if add_cc_env_for_linux:
        merged_env = select({
            # Linux platforms require that CC be set to clang.
            "@platforms//os:linux": dicts.add([{"CC": "clang"}, env]),
            "//conditions:default": env,
        })
    else:
         merged_env = env

    sh_test(
        env = merged_env
    )

```

Reply:
Sorry. What do you think about the above? Would it address your initial request and cut down on what others may need to specify?

Reply:
It would address the issue, in terms of using `bazel_integration_test` will not require `clang` anymore. But if you ask me, it looks unclear from the rule's interface perspective and the more clean way to avoid passing `CC=clang` in every rule instantiation explicitly is to locally define a macro which sets `env={"CC": "clang"}` and bypass all of the arguments to `bazel_integration_test`

Reply:
I think that you are correct that setting the value in the `bazel_integration_test` macro is weird.

It just occurred to me that we really should be getting this value from the original Bazel invocation. I think that we should add `CC` to [`_DEFAULT_ENV_INHERIT`](https://github.com/bazel-contrib/rules_bazel_integration_test/blob/main/bazel_integration_test/private/bazel_integration_test.bzl#L15). Then, all of the `CC=clang` nonsense goes away in `rules_bazel_integration_test`. What do you think?